src/nccl_ofi_net.c

```diff
@@ -27,6 +27,9 @@ bool local_mr = false;
 bool hmem_mr = false;
 /* Indicates if GPUDirect is supported by libfabric provider */
 bool support_gdr = true;
+/* Indicates if the cudaDeviceFlushGPUDirectRDMAWrites function should be used
+ * to flush data to the GPU */
+bool cuda_flush = false;
```

Comment:
Let's add a static assert that `cuda_flush` is set false in the beginning. If accidentally, a programmer sets it to true here, then the program would fail later in ofi_iflush().

Reply:
I don't think we can use a static_asset for this in C (https://stackoverflow.com/questions/33806420/how-to-use-const-int-in-static-assert). I can do a runtime assertion at initialization time, but I don't think that accomplishes what you were after. What do you think? I can add a strongly worded comment?

Reply:
There is a possibility to use `_Static_assert` as mentioned here: https://gcc.gnu.org/wiki/C11Status. Does that work?

Reply:
Here's what I'm seeing for both flavors of static assert. I guess the presence of a variable (whether const or non-const) is viewed as not being a static expression in C.
```
nccl_ofi_net.c:35:27: error: expression in static assertion is not constant
   35 | _Static_assert(cuda_flush == false);
      |                           ^
```

Reply:
Something like this would work:
```
#define CUDA_FLUSH_DEFAULT false
_Static_assert(CUDA_FLUSH_DEFAULT == false);
bool cuda_flush = CUDA_FLUSH_DEFAULT;
#undef CUDA_FLUSH_DEFAULT
```

Reply:
Ahh :-/ Thanks for trying out the options. I think it would be best to add a strongly worded comment about the default value.

Reply:
I updated the comment in one of my earlier updates. Could you let me know if it looks ok now or if there are some changes you'd like to see?